pkg/controllers/commontest/common.go

```diff
@@ -61,11 +62,14 @@ func HasOwnerRef(meta metav1.ObjectMeta, kind, name string) bool {
 	return false
 }
 
-func HasFieldOwnership(meta metav1.ObjectMeta, mgr, rawFields string) bool {
+func HasFieldOwnership(meta metav1.ObjectMeta, mgr, expected string) string {
 	for _, ref := range meta.ManagedFields {
-		if ref.Manager == mgr && string(ref.FieldsV1.Raw) == rawFields {
-			return true
+		if ref.Manager == mgr {
+			if diff := cmp.Diff(string(ref.FieldsV1.Raw), expected); diff != "" {
+				return fmt.Sprintf("(-got, +want)\n%s", diff)
+			}
+			return ""
 		}
 	}
-	return false
+	return fmt.Sprintf("No managed fields managed by %s", mgr)
```

Comment:
`HasFieldOwnership` used to return a boolean but that made it hard to debug since we could not know what was wrong with the test. So instead of a plane boolean value, I've modified it to return a diff.